src/main/java/plugily/projects/villagedefense/arena/Arena.java

```diff
@@ -454,7 +454,10 @@ public void setWave(int wave) {
   }
 
   public void spawnVillager(Location location) {
-    addVillager(CreatureUtils.getCreatureInitializer().spawnVillager(location));
+    Villager villager = CreatureUtils.getCreatureInitializer().spawnVillager(location);
+    villager.setCustomNameVisible(plugin.getConfigPreferences().getOption(ConfigPreferences.Option.NAME_VISIBLE_VILLAGER));
+    villager.setCustomName(CreatureUtils.getRandomVillagerName());
```

Comment:
Check if the custom name is visible before setting the name.

Reply:
If it's false, the player can still see the name when he looks at the villager. It's intended.

Reply:
I don't think so? If `setCustomNameVisible` was set to false, setting a new custom name to an entity will not display the name above the entity if the player is looking at.

Reply:
![2021-05-29_16 54 01](https://user-images.githubusercontent.com/25300232/120066019-a75a2280-c09e-11eb-896e-7fc7b82594be.png)
![2021-05-29_16 54 05](https://user-images.githubusercontent.com/25300232/120066021-ade89a00-c09e-11eb-8222-94c93d4cfc19.png)
![2021-05-29_16 54 09](https://user-images.githubusercontent.com/25300232/120066023-b04af400-c09e-11eb-994c-f463b1254600.png)
I don't know what happened in legacy versions, but here is what I tested in 1.16.5

Reply:
And this is Vanilla 1.16.5, no mods